.travis-ci.sh

```diff
@@ -16,15 +17,61 @@ case "$TARGET" in
       ./configure -prefix ~/ocaml
       make world.opt
       make install
-      cd ..
+      cd ../..
       rm -rf src
       echo "$OCAML_VERSION.$OCAML_RELEASE" > ~/ocaml/cached-version
     fi
+    echo -en "travis_fold:end:ocaml\r"
+    if [ $WITH_OPAM -eq 1 ] ; then
+      echo -en "travis_fold:start:opam.init\r"
+      if [ "$TRAVIS_OS_NAME" = "osx" ] ; then
+        brew install aspcud
+        PREFIX=/Users/travis
+      else
+        PREFIX=/home/travis
+      fi
+      if [ ! -e ~/ocaml/bin/opam -o ! -e ~/.opam/lock -o "$OPAM_RESET" = "1" ] ; then
+        mkdir ~/ocaml/src
+        cd ~/ocaml/src
+        wget https://github.com/ocaml/opam/releases/download/1.2.2/opam-full-1.2.2.tar.gz
+        tar -xzf opam-full-1.2.2.tar.gz
+        cd opam-full-1.2.2
+        ./configure --prefix=$PREFIX/ocaml
+        make lib-ext
+        make all
+        make install
+        cd ../..
```

Comment:
For a slightly cleaner script you could use `pushd` and `popd`.

Reply:
To me, the only time it's useful is when you need to change to a directory from an unknown starting directory and later return to it, when pushd/popd is much neater than hacks involving pwd.

I don't think it's cleaner per se (portability of course doesn't matter here). I personally find matching up pairs of pushd/popd to be considerably more error prone than looking at where in a directory tree the script is at any given line.